framework/src/problems/ReferenceResidualProblem.C

```diff
@@ -114,6 +135,34 @@ ReferenceResidualProblem::ReferenceResidualProblem(const InputParameters & param
 
   _accept_mult = params.get<Real>("acceptable_multiplier");
   _accept_iters = params.get<int>("acceptable_iterations");
+
+  const auto norm_type_enum =
+      params.get<MooseEnum>("normalization_type").getEnum<NormalizationType>();
+  if (norm_type_enum == NormalizationType::LOCAL_L2)
+  {
+    _norm_type = DISCRETE_L2;
+    _local_norm = true;
+  }
+  else if (norm_type_enum == NormalizationType::GLOBAL_L2)
+  {
+    _norm_type = DISCRETE_L2;
+    _local_norm = false;
+  }
+  else if (norm_type_enum == NormalizationType::LOCAL_LINF)
+  {
+    _norm_type = DISCRETE_L_INF;
+    _local_norm = true;
+  }
+  else if (norm_type_enum == NormalizationType::GLOBAL_LINF)
+  {
+    _norm_type = DISCRETE_L_INF;
+    _local_norm = false;
+  }
+  else
```

Comment:
Could do `switch case` here

Reply:
Why or why not? Happy to learn the better way, just didn't think it mattered much here since it's a small list in the constructor.

Reply:
This is a suggestion, not a required change. In the old days a compiler might optimize switch case more than series of if and else-if, but compilers are pretty smart now. There's no performance concern here. It's just a style preference for me; I like switch case for enums